src/konserve/impl/storage_layout.cljc

```diff
@@ -0,0 +1,93 @@
+(ns konserve.impl.storage-layout
+  "One of these protocols must be implemented by each store to provide low level
+  access depending on the low-level storage layout chosen. Stores can support
+  multiple layouts."
+  (:require [konserve.serializers :refer [serializer-class->byte byte->key]]
+            [konserve.compressor :refer [compressor->byte byte->compressor]]
+            [konserve.encryptor :refer [encryptor->byte byte->encryptor]])
+  #?(:clj (:import [java.nio ByteBuffer])))
+
+(def ^:const header-size 20)
+
+(defn create-header
+  "Return Byte Array with following content
+     1th Byte = Storage layout used
+     2th Byte = Serializer Type
+     3th Byte = Compressor Type
+     4th Byte = Encryptor Type
+   5-8th Byte = Meta-Size
+  9th-20th Byte are spare"
+  [version serializer compressor encryptor meta]
+  #?(:clj
+     (let [serializer-id        (get serializer-class->byte (type serializer))
+           compressor-id        (get compressor->byte compressor)
+           encryptor-id         (get encryptor->byte encryptor)
+
+           env-array        (byte-array [version serializer-id compressor-id encryptor-id])
+           return-buffer    (ByteBuffer/allocate header-size)
+           _                (.put return-buffer env-array)
+           _                (.putInt return-buffer 4 meta)
+           return-array     (.array return-buffer)]
+       (.clear return-buffer)
+       return-array)
+     :cljs (throw (ex-info "Not supported yet." {}))))
+
+(defn parse-header
+  "Inverse function to create-header. serializers are a map of serializer-id to
```

Comment:
*Serializers

Reply:
I was using this here to explicitly refer to the argument names, so `serializers` is written in exactly the same way as in the argument vector. To my knowledge there is no markup language similar to Java docs where arguments can be highlighted in the doc string. It might be possible though to do something like this in combination with spec.